pkg/cloud/services/ec2/eips.go

```diff
@@ -121,7 +124,8 @@ func (s *Service) releaseAddresses() error {
 				AssociationId: ip.AssociationId,
 			})
 			if err != nil {
-				return errors.Errorf("failed to release elastic IP %q with allocation ID %q: Still associated with association ID %q", *ip.PublicIp, *ip.AllocationId, *ip.AssociationId)
+				record.Warnf(s.scope.AWSCluster, "FailedDisassociateEIP", "Failed to disassociate Elastic IP %q: %v", *ip.AllocationId, err)
+				return errors.Errorf("failed to disassociate Elastic IP %q with allocation ID %q: Still associated with association ID %q", *ip.PublicIp, *ip.AllocationId, *ip.AssociationId)
```

Comment:
Not directly related to this PR, but it seems really odd that we are disassociating here, but also calling the `disassociateAddress` method below in the error block for the call to `s.scope.EC2.ReleaseAddress` below...  I'm wondering if we need to replace the call to `s.scope.EC2.DisassociateAddress` here to a call to `disassociateAddress` here, or just remove it and let it be handled below with the error handling when trying to release the address?

Reply:
I mentioned that it looked odd to @vincepri and he said there was a reason for it (which I've not forgotten)... help?

Reply:
I could see issues related to the eventual consistency of the API, but that could be handled by just retrying the call to release. There is also the case for the EIP associated with the NAT gateway, which we cannot disassociate or release until some background processes complete, but we could handle that by a call w/ retries to `disassociateAddress` above the call to `s.scope.EC2.ReleaseAddress`

Reply:
should we open a new issue to track what @detiber raised as a potential issue?  Or would you like to resolve the issues mentioned above in this PR?

Reply:
Followup issue sounds good.

Reply:
Follow up issue #1459 opened.